### ISSUE TITLE
Cmdlets not working

### ISSUE THREAD

**Issue by zalokhan** [CorfuDB/CorfuDB#280]
I am unable to run any of the cmdlets.
On trying to execute the query cmdlet I get the following error:
```
./corfu_layouts query -c localhost:9000
WARNING: cat already refers to: #'clojure.core/cat in namespace: net.cgrand.parsley.fold, being replaced by: #'net.cgrand.parsley.fold/cat
Exception in thread "main" java.io.FileNotFoundException: /Users/zlokhandwala/Documents/CorfuDB-zalokhan/bin/../corfu_scripts/corfu_layouts.clj (No such file or directory)
       	at java.io.FileInputStream.open0(Native Method)
       	at java.io.FileInputStream.open(FileInputStream.java:195)
       	at java.io.FileInputStream.<init>(FileInputStream.java:138)
       	at clojure.java.io$eval3600$fn__3601.invoke(io.clj:229)
       	at clojure.java.io$eval3491$fn__3492$G__3480__3499.invoke(io.clj:69)
       	at clojure.java.io$eval3618$fn__3619.invoke(io.clj:258)
       	at clojure.java.io$eval3491$fn__3492$G__3480__3499.invoke(io.clj:69)
       	at clojure.java.io$fn__3558.invokeStatic(io.clj:165)
       	at clojure.java.io$fn__3558.invoke(io.clj:165)
       	at clojure.java.io$eval3491$fn__3505$G__3476__3512.invoke(io.clj:69)
       	at clojure.java.io$reader.invokeStatic(io.clj:102)
       	at clojure.java.io$reader.doInvoke(io.clj:86)
       	at clojure.lang.RestFn.invoke(RestFn.java:410)
       	at clojure.lang.AFn.applyToHelper(AFn.java:154)
       	at clojure.lang.RestFn.applyTo(RestFn.java:132)
       	at clojure.core$apply.invokeStatic(core.clj:648)
       	at clojure.core$apply.invoke(core.clj:641)
       	at clojure.core$slurp.invokeStatic(core.clj:6664)
       	at clojure.core$slurp.doInvoke(core.clj:6657)
       	at clojure.lang.RestFn.invoke(RestFn.java:410)
       	at org.corfudb.shell$_formify_file_base.invokeStatic(shell.clj:41)
       	at org.corfudb.shell$_formify_file_base.invoke(shell.clj:41)
       	at org.corfudb.shell$_formify_file.invokeStatic(shell.clj:47)
       	at org.corfudb.shell$_formify_file.invoke(shell.clj:46)
       	at org.corfudb.shell$_main.invokeStatic(shell.clj:75)
       	at org.corfudb.shell$_main.doInvoke(shell.clj:50)
       	at clojure.lang.RestFn.invoke(RestFn.java:482)
       	at clojure.lang.Var.invoke(Var.java:401)
       	at clojure.lang.AFn.applyToHelper(AFn.java:171)
       	at clojure.lang.Var.applyTo(Var.java:700)
       	at clojure.core$apply.invokeStatic(core.clj:646)
       	at clojure.core$apply.invoke(core.clj:641)
       	at org.corfudb.shell_main$_main.invokeStatic(shell_main.clj:6)
       	at org.corfudb.shell_main$_main.doInvoke(shell_main.clj:4)
       	at clojure.lang.RestFn.applyTo(RestFn.java:137)
       	at org.corfudb.shell_main.main(Unknown Source)
```

**Comment by Maithem** [comment]
@no2chem If we are not planning a fix soon, i think its best we roll back. Its kind of annoying to checkout a broken master.

**Comment by slfritchie** [comment]
I'll take a look at this, since CLI stuff overlaps with test refactoring.

**Comment by Maithem** [comment]
@slfritchie maybe revert to the old cmdlts until this work is done?

**Comment by no2chem** [comment]
Fixed by #280

**Comment by slfritchie** [comment]
Not all of the cmdlets are fixed by #313 (`corfu_as` isn't working, sorry), but the remaining warts are left for future work.

**Comment by no2chem** [comment]
corfu_as was fixed by #323. What "warts" are "left for future work"?

**Comment by slfritchie** [comment]
My raw notes from today are below.  Items include (in no particular order):

* New bug #322 related to `corfu_management_bootstrap`
* No usage summary for `corfu_ping`
* `corfu_ping` now only pings once rather than series that's interruptible by control-c.
* All 0/non-0 exit status indicators from the prior cmdlet implementation in Java are gone
* Inconsistent use of `-c` flag to specify endpoint names
* `corfu_layouts edit` is hitting a bogus `WRONG_EPOCH` error (though it probably isn't the script's fault)
* Inconsistent output in success case: some scripts output `ACK`, others do nothing.
* Further insight (perhaps) on #273 based on `corfu_logunit` ad hoc tests


    bin/corfu_smrobject
            usage
            
    bin/corfu_smrobject -i 42 -c localhost:8000 org.corfudb.runtime.collections.SMRMap get not-exist
    	nil
    
    bin/corfu_smrobject -i 42 -c localhost:8000 org.corfudb.runtime.collections.SMRMap put foo bar
    	nil
    
    bin/corfu_smrobject -i 42 -c localhost:8000 org.corfudb.runtime.collections.SMRMap put foo bar2
    	bar
    
    bin/corfu_smrobject -i 43 -c localhost:8000 org.corfudb.runtime.collections.SMRMap put foo bar2
    	nil
    
    bin/corfu_smrobject -i 43 -c localhost:8000 org.corfudb.runtime.collections.SMRMap keySet
    	 #object[java.util.HashMap$KeySet 0x3c9118b8 [foo]]
    
    
    bin/corfu_as
            usage
            
    echo yodel | bin/corfu_as -i 2 -c localhost:8000 write 1
    	 CompilerException java.lang.RuntimeException: Unable to resolve symbol: get-address-space-view in this context, compiling:(NO_SOURCE_PATH:1:1467) 
    bin/corfu_as -i 2 -c localhost:8000 read 7
     CompilerException java.lang.RuntimeException: Unable to resolve symbol: get-address-space-view in this context, compiling:(NO_SOURCE_PATH:1:1467) 
    
    bin/corfu_layout
    	usage
            
    bin/corfu_layout localhost:8000
    	Query localhost:8000:
    	Layout(layoutServers=[localhost:8000], sequencers=[localhost:8000], segments=[Layout.LayoutSegment(replicationMode=CHAIN_REPLICATION, start=0, end=-1, stripes=[Layout.LayoutStripe(logServers=[localhost:8000])], replexes=null)], epoch=0)
    
    bin/corfu_layouts
            usage
            
    bin/corfu_layouts -c localhost:8000 query
    	JSON thingie
    
    bin/corfu_layouts -c localhost:8000 edit
    	starts vi, whee!
    	epoch=1, added localhost:8001 to layout server list.
    	server 8000: WARN  [event-5] o.c.i.LayoutServer - Got a request to bootstrap a server which is already bootstrapped, rejecting!
    	server 8001: not running
    		14:15:29.831 INFO  [nREPL-worker-0] o.c.r.CorfuRuntime - Connecting to Corfu server instance, layout servers=[localhost:8000]
     AnnotatedConnectException Connection refused: localhost/127.0.0.1:8001  sun.nio.ch.SocketChannelImpl.checkConnect (SocketChannelImpl.java:-2)
    
    bin/corfu_layouts -c localhost:8000 edit
    	epoch=1, added localhost:8001 to single segment
    14:17:22.072 INFO  [nREPL-worker-0] o.c.r.CorfuRuntime - Connecting to Corfu server instance, layout servers=[localhost:8000]
     Trying install with rank 0
    WrongEpochException Wrong epoch. [expected=0]  org.corfudb.runtime.clients.BaseClient.handleWrongEpoch (BaseClient.java:144)
        
    bin/corfu_layouts -c localhost:8000 edit
    	epoch=1, no other changes
    14:18:34.702 INFO  [nREPL-worker-0] o.c.r.CorfuRuntime - Connecting to Corfu server instance, layout servers=[localhost:8000]
     Trying install with rank 0
    WrongEpochException Wrong epoch. [expected=0]  org.corfudb.runtime.clients.BaseClient.handleWrongEpoch (BaseClient.java:144)
    
    
    bin/corfu_logunit
            usage
            
    bin/corfu_logunit localhost:8000 read 0
    	ClassCastException org.corfudb.protocols.logprotocol.SMREntry cannot be cast to [B  org.corfudb.shell/read-logunit (NO_SOURCE_FILE:1)
    
    bin/corfu_logunit localhost:8000 read 40
    	#object[org.corfudb.protocols.wireprotocol.DataType 0x2533e51f EMPTY]
    
    echo "scribble at 41" | bin/corfu_logunit localhost:8000 write 41
    	no output
    
    echo "scribble at 41" | bin/corfu_logunit localhost:8000 write 41
    	SERVER-SIDE:
     14:35:46.541 ERROR [event-14] o.c.i.l.StreamLogFiles - Disk_write[41]: Exception
     org.corfudb.runtime.exceptions.OverwriteException: null
    	at org.corfudb.infrastructure.log.StreamLogFiles.append(StreamLogFiles.java:303)
    14:35:46.541 ERROR [event-14] o.c.i.NettyServerRouter - Exception during read!
     java.lang.RuntimeException: org.corfudb.runtime.exceptions.OverwriteException
    	at org.corfudb.infrastructure.log.StreamLogFiles.append(StreamLogFiles.java:308)
    Caused by: org.corfudb.runtime.exceptions.OverwriteException: null
    	at org.corfudb.infrastructure.log.StreamLogFiles.append(StreamLogFiles.java:303)
    	CLIENT-SIDE:
    TimeoutException Timeout after 5000 ms  org.corfudb.util.CFUtils.lambda$failAfter$0 (CFUtils.java:108)
    
    bin/corfu_logunit localhost:8000 read 41
    	scribble at 41
    
    bin/corfu_management_bootstrap
            usage
            
    rm -f /tmp/foo ; sh -c 'bin/corfu_layouts -c localhost:8000 query 2>&1 | grep -v parsley.fold/cat | sed 1d ' > /tmp/foo
    bin/corfu_management_bootstrap -c localhost:8000 -l /tmp/foo
    	no output, but does stuff on the server side, whee
    
    bin/corfu_ping     
    	No USAGE statement
    
    bin/corfu_ping localhost:4
    PING localhost:4
    AnnotatedConnectException Connection refused: localhost/127.0.0.1:4  sun.nio.ch.SocketChannelImpl.checkConnect (SocketChannelImpl.java:-2)
    114 [sfritchie-m01]:CorfuDB: echo $status
    0
    
    bin/corfu_ping localhost:8000
    PING localhost:8000
    ACK time=39.587ms
    
    
    bin/corfu_query
    	usage
            
    bin/corfu_query localhost:8000
    	Query localhost:8000:
    	{:class org.corfudb.protocols.wireprotocol.VersionInfo, :jvmUsed /Library/Java/JavaVirtualMachines/jdk1.8.0_91.jdk/Contents/Home/jre/bin/java, :optionsMap #object[com.google.gson.internal.LinkedTreeMap 0x7f6d3d0e {--address=localhost, --compact=60, <port>=8000, --quickcheck-test-mode=false, --initial-token=-1, --single=true, --log-level=INFO, --memory=true, --max-cache=1000000000, --no-verify=false, --cm-poll-interval=1, --checkpoint=60}], :startupArgs org.corfudb.infrastructure.CorfuServer -m -s -d INFO 8000, :upTime 1179337}
    
    bin/corfu_sequencer
            usage
            
    bin/corfu_sequencer -c localhost:8000 latest
    	2
    
    bin/corfu_sequencer -i 42 -c localhost:8000 latest
    	1
    
    bin/corfu_sequencer -i 43 -c localhost:8000 latest
    	2
    
    bin/corfu_sequencer -i 44 -c localhost:8000 latest
    	-1
    
    bin/corfu_sequencer -i 66 -c localhost:8000 next-token 66
    	3
    
    bin/corfu_sequencer -i 66 -c localhost:8000 next-token 0
    	68
    
    
    bin/corfu_stream
    	usage
    
    bin/corfu_stream -i 42 -c localhost:8000 read
    	ClassCastException org.corfudb.protocols.logprotocol.SMREntry cannot be cast to [B  org.corfudb.shell/read-stream (NO_SOURCE_FILE:1)
    
    bin/corfu_stream -i 7766554433 -c localhost:8000 read
    	<no output except INFO>
    
    echo hunk1 | bin/corfu_stream -i 77111000 -c localhost:8000 write
    	<no output except INFO>
    echo hunk2 | bin/corfu_stream -i 77111000 -c localhost:8000 write
    	<no output except INFO>
    bin/corfu_stream -i 77111000 -c localhost:8000 read
    	hunk1
            hunk2
    
    
    bin/corfu_reset
    	usage
            ... reset *what*??
            ... oh, it restarts the entire server from scratch
    	 15:16:50.156 WARN  [event-10] o.c.i.BaseServer - Remote reset requested from client d1696d00-b975-426c-b31f-0242e85798c2
              ▄████████  ▄██████▄     ▄████████    ▄████████ ███    █▄
    	███    ███ ███    ███   ███    ███   ███    ███ ███    ███
    	[...]
             15:16:51.847 WARN  [main] o.c.i.LogUnitServer - Log unit opened in-memory mode (Maximum size=1 GB). This should be run for testing purposes only. If you exceed the maximum size of the unit, old entries will be AUTOMATICALLY trimmed. The unit WILL LOSE ALL DATA if it exits.
    	[... Messages from ManagementServer and GC ...]
    
    bin/corfu_reset localhost:8000
    	ACK
